src/lib/Sympa/DataSource/LDAP.pm

```diff
@@ -108,7 +125,13 @@ sub _load_next {
     }
 
     my @retrieved;
-    my $mesg = $self->__dsh;
+    my $mesg;
+    if ($self->{_page} and $self->{_page}->cookie) {
+        $mesg = $self->_open_operation(%options);
```

Comment:
- Why do search operation again? (I don't know if this is a appropriate way to do it.)
- Maybe we could use callback.

Reply:
> [kedas](https://github.com/ikedas) [4 days ago](https://github.com/sympa-community/sympa/pull/1733#discussion_r1381100811)
> Why do search operation again? (I don't know if this is a appropriate way to do it.)

Yes, calling "search" again is the correct way to do that -- see the example in https://metacpan.org/pod/Net::LDAP::Control::Paged 

> Maybe we could use callback.

Yes, we could use a callback, but it would be more changes, and it wouldn't be simpler, I think

Reply:
> > [kedas](https://github.com/ikedas) [4 days ago](https://github.com/sympa-community/sympa/pull/1733#discussion_r1381100811)
> > Why do search operation again? (I don't know if this is a appropriate way to do it.)
> 
> Yes, calling "search" again is the correct way to do that -- see the example in https://metacpan.org/pod/Net::LDAP::Control::Paged

For me, your code looks doing search operation twice for the first page. The example does not.

Reply:
Good catch ! although this was not the case in the first version of the code you looked at.

Reverted in bb40df3436be28e22c580cdde5496c0c3063c7d9.

Reply:
I understand. Thank you.